src/core/Akka/Dispatch/Mailboxes.cs

```diff
@@ -111,8 +120,16 @@ public Type FromConfig(string path)
             var config = _system.Settings.Config.GetConfig(path);
             var type = config.GetString("mailbox-type");
 
-            var mailboxType = Type.GetType(type);
-            return mailboxType;
+            try
+            {
+                var mailboxType = Type.GetType(type, true);
+                return mailboxType;
+            }
+            catch (TypeLoadException ex)
+            {
+                var msgException = string.Format("Could not find type '{0}'", type);
```

Comment:
Here we have the full `path`, `path + ",mailbox-type"` so we can provide that in the type load exception.
"Could not find the mailbox type ${type} provided in the configuration path ${path + ".mailbox-path"}`"
We should also have a check that ensures that the type is assignable to `Akka.Dispatch.Mailbox`, if not, then we need to throw a configuration exception saying that the provided type is not a mailbox type